gobblin-restli/gobblin-restli-utils/src/main/java/org/apache/gobblin/restli/EmbeddedRestliServer.java

```diff
@@ -69,6 +70,7 @@
  * * name - defaults to the name of the first resource in the resource collection.
  * * injector - an {@link Injector} to inject dependencies into the Rest.li resources.
  */
+@Singleton
```

Comment:
So, only one instance of this class is needed? If we pass different injector (or other params), then also it reuses the existing instance?

Reply:
Normally, if the class is marked with Singleton annotation , Guice will create only one instance of it. All other classes that depend on this one will get the same instance.

However, in this case the class is created by our provider, and not by Guice directly, so we need an extra step to make sure only one instance exists. I've updated Guice module like this:

` binder.bind(EmbeddedRestliServer.class).toProvider(EmbeddedRestliServerProvider.class).in(Singleton.class)`